latch_cli/services/cp/ldata_utils.py

```diff
@@ -52,111 +38,100 @@ class AccountInfoCurrentPayload(TypedDict):
-
-        is_parent = remaining is not None and remaining != ""
-
-        if not allow_resolve_to_parent and is_parent:
-            raise ValueError("Node cannot be resolved in this workspace")
-
-        if remaining is not None and "/" in remaining:
-            raise ValueError("Node cannot be resolved in this workspace")
-
-        return GetNodeDataResult(
-            name=final_link_target["name"],
-            type=LDataNodeType(final_link_target["type"].lower()),
-            is_parent=is_parent,
-        )
-
-    except (TypeError, ValueError) as e:
-        auth_header = get_auth_header()
-        match = auth.match(auth_header)
-        if match is None:
-            auth_type = auth_header
-        elif match["sdk"] is not None:
-            auth_type = "SDK Token"
-        else:
-            auth_type = "Execution Token"
-
-        auth_str = (
-            f"{click.style(f'Authorized using:', bold=True, reset=False)} {click.style(auth_type, bold=False, reset=False)}"
-            + "\n"
-        )
-
-        ws_id = user_config.workspace_id
-        ws_name = user_config.workspace_name
-
-        resolve_str = (
-            f"{click.style(f'Relative path resolved to:', bold=True, reset=False)} {click.style(normalized, bold=False, reset=False)}"
-            + "\n"
-        )
-        ws_str = (
-            f"{click.style(f'Using Workspace:', bold=True, reset=False)} {click.style(ws_id, bold=False, reset=False)}"
-        )
-        if ws_name is not None:
-            ws_str = f"{ws_str} ({ws_name})"
-        ws_str += "\n"
-
-        account_relative = is_account_relative(append_scheme(remote_path))
-
-        raise PathResolutionError(
-            click.style(
-                f"""
-{click.style(f'{remote_path}:', bold=True, reset=False)}{click.style(f" not found", bold=False, reset=False)}
-{resolve_str if account_relative else ""}{ws_str if account_relative else ""}
-{auth_str}
-{click.style("Check that:", bold=True, reset=False)}
-{click.style("1. The target object exists", bold=False, reset=False)}
-{click.style(f"2. Account ", bold=False, reset=False)}{click.style(acc_id, bold=True, reset=False)}{click.style(" has permission to view the target object", bold=False, reset=False)}
-{"3. The correct workspace is selected" if account_relative else ""}
-
-For privacy reasons, non-viewable objects and non-existent objects are indistinguishable""",
-                fg="red",
+    ret: Dict[str, GetNodeDataResult] = {}
+    for i, remote_path in enumerate(remote_paths):
+        node: LdataResolvePathToNodePayload = res[f"q{i}"]
+
+        try:
+            final_link_target = node["ldataNode"]["finalLinkTarget"]
+            remaining = node["path"]
+
+            is_parent = remaining is not None and remaining != ""
+
+            if not allow_resolve_to_parent and is_parent:
+                raise ValueError
+
+            if remaining is not None and "/" in remaining:
+                raise ValueError
```

Comment:
More specificity in these errors?

Reply:
I know they are handled by `get_path_error` but each of these requires a different user action, right?

Reply:
sure